plan/planbuilder.go

```diff
@@ -36,6 +36,7 @@ var (
 	ErrUnknownColumn        = terror.ClassOptimizerPlan.New(CodeUnknownColumn, "Unknown column '%s' in '%s'")
 	ErrWrongArguments       = terror.ClassOptimizerPlan.New(CodeWrongArguments, "Incorrect arguments to EXECUTE")
 	ErrAmbiguous            = terror.ClassOptimizerPlan.New(CodeAmbiguous, "Column '%s' in field list is ambiguous")
+	ErrAlterAutoID          = terror.ClassAutoid.New(CodeAlterAutoID, "No support for setting auto_increment using alter_table")
```

Comment:
I have searched in MySQL 8.0 Server Error Codes and Messages and got these error mentioned `AUTO_INCREMENT` or `AUTOINC`. In my opinon there have no appropriate MySQL standard error code to describe this error. Because this problem caused by the implementation of `autoid` module, so I put this error under `ClassAutoid`.

> Error: 1164 SQLSTATE: 42000 (ER_TABLE_CANT_HANDLE_AUTO_INCREMENT)
   Message: The used table type doesn't support AUTO_INCREMENT columns

> Error: 1467 SQLSTATE: HY000 (ER_AUTOINC_READ_FAILED)
   Message: Failed to read auto-increment value from storage engine

> Error: 1569 SQLSTATE: HY000 (ER_DUP_ENTRY_AUTOINCREMENT_CASE)
   Message: ALTER TABLE causes auto_increment resequencing, resulting in duplicate entry '%s' for key '%s'  

> Error: 1671 SQLSTATE: HY000 (ER_BINLOG_UNSAFE_AUTOINC_COLUMNS)
   Message: Statement is unsafe because it invokes a trigger or a stored function that inserts into an AUTO_INCREMENT column. Inserted values cannot be logged correctly.

> Error: 1722 SQLSTATE: HY000 (ER_BINLOG_UNSAFE_WRITE_AUTOINC_SELECT)
   Message: Statements writing to a table with an auto-increment column after selecting from another table are unsafe because the order in which rows are retrieved determines what (if any) rows will be written. This order cannot be predicted and may differ on master and the slave.

> Error: 1723 SQLSTATE: HY000 (ER_BINLOG_UNSAFE_CREATE_SELECT_AUTOINC)
   Message: CREATE TABLE... SELECT... on a table with an auto-increment column is unsafe because the order in which rows are retrieved by the SELECT determines which (if any) rows are inserted. This order cannot be predicted and may differ on master and the slave.

> Error: 1727 SQLSTATE: HY000 (ER_BINLOG_UNSAFE_AUTOINC_NOT_FIRST)
   Message: INSERT into autoincrement field which is not the first part in the composed primary key is unsafe.

> Error: 1854 SQLSTATE: HY000 (ER_ALTER_OPERATION_NOT_SUPPORTED_REASON_AUTOINC)
   Message: Adding an auto-increment column requires a lock

> Error: 1869 SQLSTATE: HY000 (ER_AUTO_INCREMENT_CONFLICT)
   Message: Auto-increment value in UPDATE conflicts with internally generated values

> Error: 3109 SQLSTATE: HY000 (ER_GENERATED_COLUMN_REF_AUTO_INC)
   Message: Generated column '%s' cannot refer to auto-increment column.